browser/templates/tool.html

```diff
@@ -71,37 +67,45 @@ <h4>{{ settings['title'] | title }}</h4>
       <!-- Second canvas used for acutal image processing, hidden -->
       <canvas id="hidden_seg_canvas" class="hide"></canvas>
     </div>
-
   </div>
 </main>
 {%- endblock -%}
 
 {%- block extraJs -%}
-{%- if settings['filetype'] == "zstack" -%}
-  <script type="text/javascript" src="{{ url_for('static', filename='js/main_zstack.js') }}"></script>
-  <script type="text/javascript" src="{{ url_for('static', filename='js/actions.js') }}"></script>
-{%- elif settings['filetype'] == "track" -%}
-  <script type="text/javascript" src="{{ url_for('static', filename='js/main_track.js') }}"></script>
-{%- endif -%}
+  <script src="{{ url_for('static', filename='js/tool.js') }}"></script>
+  <script>
+    var settings = {{ settings | tojson}};
+
+    const outputBucket = settings.output_bucket;
+    const projectID = settings.token;
+    function download() {
+      window.location = `/downloadproject/${projectID}`;
+    }
+    function upload() {
+      // disable button and show loading bar
+      document.getElementById('submit').classList.add('disabled');
+      document.getElementById('loading-bar').classList.add('active');
+      const promise = $.ajax({
+        type: 'POST',
+        url: `${document.location.origin}/upload_file/${outputBucket}/${projectID}`,
+        async: true
+      });
```

Comment:
I liked how you moved more of the HTML scripts into a JS file. Would it make sense to do that here as well?

Reply:
I initially tried moving the whole block to a JS file, but some parts depend on access to the `settings`, which are only accessible through the Flask/Jinja template as far as I'm aware. In particular, we use the settings to access the project ID and the output bucket for uploading to S3.